app/models/work_experience.rb

```diff
@@ -0,0 +1,38 @@
+# == Schema Information
+#
+# Table name: work_experiences
+#
+#  id          :integer          not null, primary key
+#  company     :string
+#  role        :string
+#  start_date  :date
+#  end_date    :date
+#  city        :string
+#  description :string
+#  profile_id  :integer
+#
+
+class WorkExperience < ApplicationRecord
+  belongs_to :profile 
+
+  validates :start_date, presence: true
+  validates :end_date, presence: true
+  validates :company, presence: true
+  validates :role, presence: true
+  validates :city, presence: true
+  validate :end_date_cannot_be_earlier_than_or_equal_to_start_date
+
+  private
+
+  def end_date_cannot_be_earlier_than_or_equal_to_start_date
+    if !end_date.nil? && !start_date.nil?
+      if end_date < start_date
+        errors.add :start_date, "Cannot later than end date."
+        errors.add :end_date, "Cannot earlier than start date."
+      elsif end_date == start_date
+        errors.add :start_date, "Cannot equal to end date."
+        errors.add :end_date, "Cannot equal to start date."
+      end
```

Comment:
```
2.6.1 :001 > a = nil
 => nil
2.6.1 :002 > a.nil?
 => true
2.6.1 :003 > !a.nil?
 => false
2.6.1 :004 > !!a
 => false
```
 You can just do `!!a` or `if end_date.present? && start_date.present?` or `if end_date && start_date`

ruby is smart enough to run the checking

```
2.6.1 :005 > if a
2.6.1 :006?>   puts "hello"
2.6.1 :007?>   end
 => nil
2.6.1 :008 > b = 123
 => 123
2.6.1 :009 > if b
2.6.1 :010?>   puts b
2.6.1 :011?>   end
123
```